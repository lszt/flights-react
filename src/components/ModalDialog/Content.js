import styled from 'styled-components';
import Centered from '../Centered';

const Content = styled(Centered)`
  position: fixed;
  border: 1px solid #eee;
  border-radius: 10px;
  padding: 1em;
  background-color: #fff;
  box-shadow: 0 0 2em 0 #aaa;
  z-index: 10000;
  
  @media(max-width: ${props => props.fullWidthThreshold}px) {
    width: 85%;
  }
`;

export default Content;
