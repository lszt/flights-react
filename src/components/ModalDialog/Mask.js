import styled from 'styled-components';

const Mask = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: #fff;
  opacity: 0.7;
  z-index: 9999;
`;

export default Mask;
