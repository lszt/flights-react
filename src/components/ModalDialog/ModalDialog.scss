.ModalDialog {
  .mask {
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background-color: #fff;
    opacity: 0.7;
    z-index: 9999;
  }

  .content {
    position: fixed;
    top: 50%;
    left: 50%;
    margin-right: -50%;
    transform: translate(-50%, -50%);
    border: 1px solid #eee;
    border-radius: 10px;
    padding: 1em;
    background-color: #fff;
    box-shadow: 0 0 2em 0 #aaa;
    z-index: 10000;
  }
}
