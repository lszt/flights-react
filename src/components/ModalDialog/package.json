{
  "name": "ModalDialog",
  "version": "0.0.0",
  "private": true,
  "main": "./ModalDialog.js"
}
