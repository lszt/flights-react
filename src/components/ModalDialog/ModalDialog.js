import React, {PropTypes, Component} from 'react';
import Mask from './Mask';
import Content from './Content';

const handleMaskClick = onBlur => {
  if (typeof onBlur === 'function') {
    onBlur();
  }
};

const ModalDialog = props => (
  <div>
    <Mask onClick={handleMaskClick.bind(props.onBlur)}/>
    <Content fullWidthThreshold={props.fullWidthThreshold}>{props.content}</Content>
  </div>
);

ModalDialog.propTypes = {
  content: PropTypes.element.isRequired,
  onBlur: PropTypes.func,
  fullWidthThreshold: PropTypes.number
};

ModalDialog.defaultProps = {
  fullWidthThreshold: 600
};

export default ModalDialog;
