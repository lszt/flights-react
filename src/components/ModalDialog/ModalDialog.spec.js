import React from 'react';
import ReactDOM from 'react-dom';
import TestUtils from 'react-dom/test-utils';
import expect from 'expect';
import { ThemeProvider } from 'styled-components';
import ModalDialog from './ModalDialog';
import Utils from '../../../test/Utils';

describe('components', () => {
  describe('ModalDialog', () => {
    it('renders given content', () => {
      const content = <div>My test content</div>;
      const modalDialog = (
        <ThemeProvider theme={{}}>
          <ModalDialog content={content}/>
        </ThemeProvider>
      );

      const component = TestUtils.renderIntoDocument(modalDialog);
      const element = ReactDOM.findDOMNode(component);

      const contentWrapper = element.childNodes[1];
      const contentNode = contentWrapper.childNodes[0];

      expect(contentNode.innerHTML).toBe('My test content');
    });

    it('calls onBlur handler on mask click', () => {
      const handler = Utils.callTracker();

      const content = <div>My test content</div>;
      const modalDialog = (
        <ThemeProvider theme={{}}>
          <ModalDialog content={content} onBlur={handler}/>
        </ThemeProvider>
      );

      const component = TestUtils.renderIntoDocument(modalDialog);
      const element = ReactDOM.findDOMNode(component);

      const mask = element.childNodes[0];

      TestUtils.Simulate.click(mask);

      const calls = handler.calls();
      expect(calls.length).toBe(1);
    });
  });
});
