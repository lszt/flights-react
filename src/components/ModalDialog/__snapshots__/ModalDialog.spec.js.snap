// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components ModalDialog renders given content 1`] = `
"<div>
  <styled.div onClick={[Function: bound handleMaskClick]} />
  <Styled(styled.div) fullWidthThreshold={600}>
    <div>
      My test content
    </div>
  </Styled(styled.div)>
</div>"
`;
